### ISSUE TITLE
Ventoy does not boot any OS

### ISSUE THREAD

**Issue by meydominic** [ventoy/Ventoy#282]
I've installed latest ventoy on my usb stick. It's working fine in general. Installed some computers with it.

But now I have an old notebook (hp pavilion dv6) from ~2011 which was shipped with Windows 7, where I only get a blinking cursor after selecting an operating system. Tried many different operating systems, but everytime I only get a blinking cursor on the upper left corner.

Do you have any ideas what I can do?

**Comment by jr1221** [comment]
I also have a computer that won’t boot any ISO’s.  It is a 2011 iMac, and I wish I could figure out what the incompatibility is.

**Comment by TeamLinux01** [comment]
At work, we use "HP Compaq dc7800 Small Form Factor PC", "HP Compaq Pro 6300 Small Form Factor PC", "HP Compaq Pro 6200 Small Form Factor PC" and "HP Compaq Pro 6000 Small Form Factor PC". I have updated to the latest BIOSes from HP, but when trying to legacy boot Ventoy (GPT formatted) on any of them, I get a blinking cursor before the menu appears.

I am sure it is a problem with the way HP bootstraps the legacy code, as I have a Dell at home and it legacy boots a GPT formatted USB drive without issue.

**Comment by tbrtje** [comment]
Hi there, just wanted to chime in.
UEFI-formatted installs of Ventoy have the same issue on my MacBook Pro 5,5, which is Efi-only. So the Legacy bootstrapping isn't an issue.

**Comment by NovaViper** [comment]
Same issue on a 2010 Mac desktop computer. I can get to the ventoy screen but once I select an OS to boot from it just shows a white underscore line in the middle of the screen

**Comment by xel-arjona** [comment]
Here testing a Vento 1.0.27, both GPT or MBR formated, with or without SecureBoot enabled on an updated (Catalina 10.15.7 firmware) MacbookPro11,3:  Same problem verified, using different images I could found a curious fenomena:  ISO's simply don't boot but some IMG files could, apparently those that use ['ELILO' booting manager](https://www.rodsbooks.com/efi-bootloaders/elilo.html):  OpenELEC-Generic.x86_64-8.0.4.img or Lakka-Generic.x86_64-2.3.2.img (both linux for Kodi) actually boots kernel but could not mount it's root,  only img that boot without problems was [UEFI's free commercial version of memtest86-usb.img](https://www.memtest86.com/downloads/memtest86-4.3.7-usb.img.zip).

Hope this help for developers to find what's going on here...

Cheers.

**Comment by thomasjeung** [comment]
I tried the same Ventoy USB stick on thinkpads t440,x240 and x250. Almost the same bios settings for all those because I use Hackintosh but noticed only on x250, Ventoy was not able to boot any iso on it. I had Arch, Fendora, windows and ubuntu on the usb stick but none of those Iso worked.

**Comment by chemicalwill** [comment]
@NovaViper did you end up getting anywhere with this? I'm having the same issue

**Comment by NovaViper** [comment]
> @NovaViper did you end up getting anywhere with this? I'm having the same issue. Have tested on multiple computers with both MBR and GPT options 

No, I eventually have up on flashing with ventoy and used etcher for the Mac, in every other device however I use ventoy

**Comment by CityguyUSA** [comment]
Can't boot on Asus G73Jw with UEFI set in BIOS.

**Comment by justcruisinround** [comment]
I tried Ventoy for the first time yesterday...encountered similar problem whereby several attempts at different ISOs did not work; just blank screen with blinking cursor. Later I discovered a trick (https://www.ventoy.net/en/doc_compatible_mark.html) to copy and paste ventoy.dat onto the usb's root directory of the image(ISO) file(s), and now able to boot...

**Comment by steve6375** [comment]
That page says
`'Just put a file with name ventoy.dat or VENTOY.DAT in the root directory of the iso image (inside of the ISO). The content of the file doesn't matter, however this file (ventoy.dat) is recommended.'`

Note that it says 'INSIDE THE ISO' - it does not say anything about placing the file the root directory???

Are you sure that fixed the problem - or could you use any name for the file or any file or any file in any folder?

**Comment by UlisseMini** [comment]
> Here testing a Vento 1.0.27, both GPT or MBR formated, with or without SecureBoot enabled on an updated (Catalina 10.15.7 firmware) MacbookPro11,3: Same problem verified, using different images I could found a curious fenomena: ISO's simply don't boot but some IMG files could, apparently those that use ['ELILO' booting manager](https://www.rodsbooks.com/efi-bootloaders/elilo.html): OpenELEC-Generic.x86_64-8.0.4.img or Lakka-Generic.x86_64-2.3.2.img (both linux for Kodi) actually boots kernel but could not mount it's root, only img that boot without problems was [UEFI's free commercial version of memtest86-usb.img](https://www.memtest86.com/downloads/memtest86-4.3.7-usb.img.zip).
> 
> Hope this help for developers to find what's going on here...
> 
> Cheers.

I have the same problem on my 2016 macbook pro. all my iso linux distros don't work but tails (a .img) does
```
uli@mbp ~/isos]$ file *
archlinux-2021.12.01-x86_64.iso:                         ISO 9660 CD-ROM filesystem data (DOS/MBR boot sector) 'ARCH_202112' (bootable)
manjaro-xfce-21.2.0-211220-linux515.iso:                 ISO 9660 CD-ROM filesystem data (DOS/MBR boot sector) 'MANJARO_XFCE_2120' (bootable)
nixos-minimal-21.11.334797.6979c0e49bb-x86_64-linux.iso: ISO 9660 CD-ROM filesystem data (DOS/MBR boot sector) 'nixos-minimal-21.11-x86_64' (bootable)
Parrot-security-4.11.3_amd64.iso:                        ISO 9660 CD-ROM filesystem data (DOS/MBR boot sector) 'Parrot security 4.11.3' (bootable)
tails-amd64-4.25.img:                                    DOS/MBR boot sector; partition 1 : ID=0xee, start-CHS (0x0,0,2), end-CHS (0x3ff,255,63), startsector 1, 2332671 sectors

```
I feel like this is key to fixing the problem. I'm investigating this but I don't know much about boot processes so any help is appreciated!

**Comment by FlyinPancake** [comment]
I have this issue on my 2011 mid mac mini. I've tried windows10 21H2, Pop_OS21.10, ubuntu and so on. Using rufus all ISOs work

**Comment by CityguyUSA** [comment]
I was able to accomplish inconsistent boots but it was locked up after the boot.  Seems to me this is best left alone.

**Comment by Drugwash2** [comment]
Just adding to the pile of complaints.
Trying to boot a **HP dc7100 CMT** and a **HP dc7800 SFF** in order to test/install some Linux distros.
Both machines have had their BIOS updated to last available versions.
None of them would boot Ventoy [either 1.64, 1.65, 1.70] off of a 32GB **Kingston Data Traveler 3.0** USB stick **MBR** formatted.
Same failure with a no-name older 4GB USB stick either MBR or GPT formatted.
Screen is black, cursor blinks top-left, nothing else happens.
Tried changing whatever seemed related in BIOS, according to HP Support and various sites, to no avail.
Tried the precise step procedure* mentioned on some sites, to no avail.
Admittedly the HP BIOSes in those machine series is quite anal but still there should be a visible attempt.
What is to be done...?

*Turn off computer, remove USB stick, turn on computer, press F9 for boot menu, insert USB stick, select USB drive.

**Comment by steve6375** [comment]
Does it boot on other systems?
Are you legacy booting or UEFI booting?

**Comment by Drugwash2** [comment]
The 32GB flash did boot on this Samsung R-580 notebook.
The dc7800 does boot with DVDs (but not CDs, possibly bad optical unit) and curently has latest Mageia on HDD.
None of the two machines has UEFI options unless I'm badly mistaken. Everything points to legacy.

**Comment by steve6375** [comment]
*Turn off computer, remove USB stick, turn on computer, press F9 for boot menu, insert USB stick, select USB drive.

????
It should be ...

*Turn off computer, insert USB stick, turn on computer, press F9 for boot menu, , select USB drive.
https://youtu.be/_zfYpezIJ2E

**Comment by Drugwash2** [comment]
See here for one: https://serverfault.com/a/1015987
Anyway I tried every possible way. If I do it your way it won't even display a BIOS boot menu, it'll go directly to first item (which is USB) and gets stuck at the flashing cursor.

BTW the 4GB flash formatted as GPT just booted an ASUS M2N-CM DVI board.

**Comment by steve6375** [comment]
Well, if you tried 'EVERY POSSIBLE WAY' then there is no solution in the world!
I suspect that you have not tried every possible way. The thread you posted has several different things to try including using an earlier BIOS, resetting BIOS to defaults, using F5, etc,
```

- disable (F5) and re-enabling (F5 again) "Hard Drive" under "Legacy Boot Sources" in the Boot Order seems to make the USB key available as a boot device under the F9 menu

- unplug the USB key, rebooting, then plugging the USB key back in results in the USB key no longer being available in the F9 menu, so not being able to boot off it. Disabling and re-enabling the Hard Drive as a boot source, as in the step above, is then required in order to see the USB key in the F9 menu again

```

Note that when booting from USB, if legacy booting then I would use an MBR partitioned USB drive not GPT. You may also like to rename (temporarily) the \EFI folder on partition 2 of the Ventoy drive to some other name to prevent UEFI booting (e.g. \EFIxxx).
You could also try other types of multiboot sticks (no need to add ISOs) - e.g. YUMI, Easy2Boot, WinSetupFromUSB, SARDU, etc, and see if they also fail in a similar way. If they do then you know the problem is not a Ventoy issue but an issue with your PCs BIOSes which you need to try harder to solve.

**Comment by steve6375** [comment]
Found this - seems improbably but you can try it:
'Yes, I too had same problem with my dc7800. I entered into bios set up: -> security-> system security-> **data execution prevention** : Enabled -> changed to **disabled**.'

Also
We have 1000+ identical desktops, and normally we're ghosting desktops with PXE and WoL server, but if anyone tries to boot with bootable USB, I will recommend that you disable in Bios(F10)/Storage/Boot order: "EFI-Boot Sources" (F5-disable);(F10-save change);   and under Bios/Storage/Storage Options you have to change the last option "hard drive.

**Comment by Drugwash2** [comment]
https://www.bleepingcomputer.com/forums/t/579505/at-wits-end-hp-dc7800-sff-will-not-boot-from-usb/
https://h30434.www3.hp.com/t5/Desktop-Boot-and-Lockup/HP-DC7800-refuses-to-boot-from-USB/td-p/6412913
The above, found when searching for solutions, are proof that the HP BIOS is badly designed. Even so, once in a blue moon someone managed to make it work. I haven't, although fixing computer hardware and software has been a hobby of mine for over twenty years. Bad luck? Bad BIOS? Dunno, the guys above seem to have stumbled into the same or similar situation.
Since my only working machine is running Linux (Mint) it'll be quite hard for me to try all those Windows applications you mentioned. Would've gone for Rufus, mentioned by one of the users as having eventually worked on a dc7800.

Fiddling with the BIOS is risky and I'm no longer young and eager to experiment. I did at least try with the original BIOS versions prior to updating them and still wouldn't boot from the Ventoy USB, at least not the dc7800. Did reset to defaults, did clear the BIOS through jumper and button, did disable/enable HDD through F5 and all that.
Renaming the EFI folder to EFI000 didn't help either. Maybe the USB boot partition not being the first one could be a/the factor?
No idea, and I'm tired after 24h of non-stop tinkering with this darn hardware and software. Maybe I'll dream the solution.

**Comment by steve6375** [comment]
Maybe you know someone with a Windows system and you can make some alternative boot sticks? Winsetupfromusb uses Microsoft Legacy boot files (MBR + bootmgr) and should be the most compatible or maybe a Rufus + Win10 stick.
Standard BIOS settings most systems are:
Fast Boot disabled
Secure Boot disabled
CSM Enabled
Also, try all USB ports (switch off - insert usb - switch on) and make sure all other USB ports (incl. internal USb devices such as card reader if fitted) are disconnected (try to use PS/2 kbd if you have one, otherwise you have to use USB kbd).

**Comment by Drugwash2** [comment]
PS/2 keyboard and mouse, no other USB device external or internal. Tried 5 out of 6 rear USB ports so far. The 2 front ports don't work for some reason, there's voltage but no device gets detected there.

Unfortunately got no other spare sticks and no access to Windows other than a 8.1 and maybe an XP. No easy way to transfer files between Mint and any of the Windows (except for using the sticks themselves), and no common access to Internet - need to move HSDPA modem from Mint to the other machine(s).

Honestly, old 486/Pentium generic BIOS versions were much more resiliant and compliant despite their myriad of settings. Proprietary stuff is bad and limited. And they call it 'progress'.
Going to sleep now. Thanks for the help.

**Comment by Drugwash2** [comment]
Tests on HP dc7800 SFF in Linux Mint 19.2 x64:
- **multibootusb 9.2.0**, 4GB disk NTFS => **boot**, winpup v2 runs, Peppermint and MX 21 XFCE don't launch
-- under Wine => can't detect any USB drive, crashes when selecting ALL drives, ditched
- **unetbootin 681**, 4GB disk FAT32 => **boot**, Peppermint runs
-- under Wine => can't detect any USB drive, ditched
- **SARDU 4**, 4GB disk FAT32 => too stupid to list more than one USB disk (either NTFS or FAT32), **can't use**, ditched
-- under Wine => can't run from path with spaces - in 2022 !!!, ditched
- **Easy2Boot (E2B)** => home page messy, unclear instructions for Linux, **can't install**, ditched
-- under Wine => can't detect any USB drive, ditched
- **Rufus 3.17p** under Wine => **can't detect** any USB drive, ditched
- **RMPrepUSB Portable 2.1.745** under Wine => **can't detect** any USB drive, ditched
- **WinSetupFromUSB 1.10** under Wine => **can't detect** any USB drive, ditched
- **YUMI 2.0.9.4** under Wine => **erroneous detection** of drives (NTFS instead of FAT32) when building disk, **won't boot**

From the little observed that machine will only accept plain MBR disks, meaning no GPT or hybrid MBR, and it will happily boot with SYSLINUX from either FAT32 or NTFS.

**Comment by steve6375** [comment]
With sardu, why not just remove all usb devices except for the one you want?
For e2b, there is a Linux script provided. Just format as MBR NTFS first.
https://easy2boot.xyz/create-your-website-with-blocks/make-using-linux/

**Comment by Drugwash2** [comment]
Let's say the other USB devices are busy at the moment; should I stop everything I do because of a poorly designed software?

The E2B site is so badly designed that important information is easy to overlook. I see now the scripts are buried deep into a subfolder. But still there are too many hoops to jump through compared to all other tools.

Anyway, on a whim I just installed **FreeNAS 9.2.1.9* (last MBR version) to the same 4GB USB disk using Mint's built-in tool **Disk Image Writer**, and the HP booted from it right away; it's currently running but too weird to configure for usage.

Bottom line is, the combination of FAT16 + exFAT + GPT used by Ventoy is more than likely never gonna be accepted by that machine and probably other close models. Sad thing is none of the tested tools provides a true and reliable multiboot environment usable with such machines.

**Comment by steve6375** [comment]
Ventoy can make an MBR usb disk, just use the options tab.

**Comment by Drugwash2** [comment]
I did not only once. No dice.
And I did reset the disk properly to MBR using **gdisk** after being set to GPT, as GParted can't figure it out.
So it must be the rest: exFAT+FAT16. Or something related to that, since MBR+NTFS and MBR+FAT32 are both accepted.

**Comment by jeremiah-miller** [comment]
I tried putting ventoy.dat into an iso and it still didn't work. I followed [these instructions](https://www.techwalla.com/articles/how-to-add-files-to-an-iso-image-in-linux) to add the file to an iso, and then mounted the resulting iso to confirm ventoy.dat was in it, and it still had the same problem

**Comment by albotroz** [comment]
The same problem.
I gave up with Ventoy. Every time it is needed there are problems with booting - after selecting an OS or other.

So still classic: USB = one OS

**Comment by steve6375** [comment]
> I tried putting ventoy.dat into an iso and it still didn't work. I followed [these instructions](https://www.techwalla.com/articles/how-to-add-files-to-an-iso-image-in-linux) to add the file to an iso, and then mounted the resulting iso to confirm ventoy.dat was in it, and it still had the same problem

That only works if the iso has been especially written to work with Ventoy. Putting that file in the iso will not MAKE an iso compatible, it must already be compatible

**Comment by ibay770** [comment]
> I have the same issue on a MacBook Pro 2011, Ventoy boots, but any ISO I select (supported, as listed) only results in a blinking cursor.

I am wondering if ventoy has the efi to boot but the iso doesn't so it is acting that way.
